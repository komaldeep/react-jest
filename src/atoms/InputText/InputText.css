.wrap {
  position: relative;
}

.input {
  height: 2.1rem;
  padding: .45rem 2.5rem .4rem .8rem;
  margin-bottom: 1.5rem;
  margin-top: 1.5rem;
}

::-ms-clear {
  display: none;
  width: 0;
  height: 0;
}

.hasDanger {
  padding-right: .8rem;
}

.disabled {

}

.pending {
  /*focus {*/
    /*box-shadow: none;*/
  /*}*/
}

.reset {
  display: inline-block;
  position: absolute;
  top: 0;
  right: 0;
  z-index: 10;
  width: .8rem;
  height: 100%;
  padding-right: 1.75rem;
  padding-left: .75rem;
  border: none;
  outline: none;
  background: transparent;
}
