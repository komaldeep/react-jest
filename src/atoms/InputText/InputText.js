import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import './InputText.css';

export default class InputText extends Component {

  static displayName = 'atoms/InputText';

  static propTypes = {
    value: PropTypes.string,
    updateValue: PropTypes.string,
    name: PropTypes.string,
    id: PropTypes.string,
    type: PropTypes.string,
    placeholder: PropTypes.string,
    handleChange: PropTypes.func,
    onKeyPress: PropTypes.func,
    maxLength: PropTypes.number,
    disableDeleteButton: PropTypes.bool,
  };

  static defaultProps = {
    type: 'text',
    placeholder: '',
    disabled: false,
    showSuccess: false,
    showError: false,
    pending: false,
    focus: false,
    maxLength: 9999,
  };

  constructor(props) {
    super(props);
    this.keepFocus = false;
    this.state = {
      value: props.value,
    };
  }

  handleOnChangeValue = () => {

  }

  render() {
    const {
      name,
      type,
      placeholder,
      disabled,
      id,
      showError,
      showSuccess,
      onKeyPress,
      maxLength,
      disableDeleteButton,
    } = this.props;

    return (
      <div className="wrap">
        <input
          type={type}
          name={name}
          id={id}
          value={this.state.value}
          placeholder={placeholder}
          ref={(input) => { this.inputField = input; }}
          className={
            classNames({
              input: true,
            })
          }
          disabled={disabled}
          onKeyPress={onKeyPress}
          maxLength={maxLength}
          onChange={this.handleOnChangeValue}
        />
      </div>
    );
  }
}
