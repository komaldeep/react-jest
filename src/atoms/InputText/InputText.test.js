import React from 'react';
import renderer from 'react-test-renderer';
import InputText from './InputText';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

test('renders without crashing', () => {
  const rendered = renderer.create(<InputText />).toJSON();
  expect(rendered).toMatchSnapshot();
});


test('inputfield with different feature', () => {
  const rendered = shallow(
    <InputText
      type= 'text'
      placeholder="placeholder for input text"
      disabled= {false}
      showSuccess= {false}
      showError= {false}
      pending= {false}
      focus= {false}
  />);
  const tree = toJson(rendered);
  expect(tree).toMatchSnapshot();
});

test('inputfield onchange event', () => {
  const wrapper = mount(
    <InputText
      type="text"
      name="name"
    />
  );
  wrapper.find('input').simulate('change', { target: {
    value: 'Change function' },
  });
  expect(wrapper.state().value).toEqual('Change function');
});
