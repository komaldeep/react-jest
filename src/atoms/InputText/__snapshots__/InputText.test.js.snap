// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inputfield with different feature 1`] = `
<div
  className="wrap"
>
  <input
    className="input"
    disabled={false}
    maxLength={9999}
    onChange={[Function]}
    placeholder="placeholder for input text"
    type="text"
  />
</div>
`;

exports[`renders without crashing 1`] = `
<div
  className="wrap"
>
  <input
    className="input"
    disabled={false}
    id={undefined}
    maxLength={9999}
    name={undefined}
    onChange={[Function]}
    placeholder=""
    type="text"
    value={undefined}
  />
</div>
`;
